### PR TITLE
[#23] [UI] As a logged-in user, I can see the NPS Survey Question

### LINKED ISSUE #23: [UI] As a logged-in user, I can see the NPS Survey Question
## Why
The user can see the question and the available answers and select their answer.

## Acceptance Criteria
- Show the question.
- Show the 1-10 score.
- User can select their score.

## Design

Screenshot 2023-03-23 at 15 02 54

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- Closes #23 

## What happened 👀

The user can see the question and the available answers and select their answer.

## Insight 📝

- Show the question.
- Show the answers as a score.
- User can select their score. When the user selects one score, the scores behind the selected score are selected too.

## Proof Of Work 📹

https://github.com/nimblehq/flutter-ic-kaung-thieu/assets/32578035/2448af95-8ad8-4125-b2a1-352ff2fefc72

### CODE REVIEW

File: lib/screens/detail/nps_answer.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                  }
+                }()),
+              ),
+              const Expanded(child: SizedBox.shrink()),
+              Text(
+                _localizations.surveyNpsExtremelyLikely,
+                style: _textTheme.labelMedium?.copyWith(color: () {
+                  if (_selectedIndex != null &&
+                      (_selectedIndex ?? 0) >=
+                          widget.answers.length / 2.round()) {
```
Comment (ducbm051291) [inline lib/screens/detail/nps_answer.dart:66]: Please check this 
```suggestion
                          (widget.answers.length / 2).round()) {
```

File: lib/screens/detail/nps_answer.dart [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
+          child: Row(
+            children: [
+              Text(
+                _localizations.surveyNpsNotAtAllLikely,
+                style: _textTheme.labelMedium?.copyWith(color: () {
+                  if (_selectedIndex != null &&
+                      (_selectedIndex ?? 0) <
+                          widget.answers.length / 2.round()) {
```
Comment (ducbm051291) [inline lib/screens/detail/nps_answer.dart:53]: Oops, this one too :D

File: lib/screens/detail/nps_answer.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+class NpsAnswerState extends State<NpsAnswer> {
+  TextTheme get _textTheme => Theme.of(context).textTheme;
+
+  AppLocalizations get _localizations => AppLocalizations.of(context)!;
+  int? _selectedIndex;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
```
Comment (ducbm051291) [inline lib/screens/detail/nps_answer.dart:27]: I think we should make a flag here and use it for below code? 

```suggestion
    final isNotAtAllLikely = (_selectedIndex != null &&
                      (_selectedIndex ?? 0) <
                          widget.answers.length / 2.round())
```

**Comment by ducbm051291** [comment]
Please check my comment and resolve conflict ;)

File: lib/screens/detail/nps_answer.dart (hunk 3 above)
Comment (kaungkhantsoe) [reply lib/screens/detail/nps_answer.dart:27]: @ducbm051291 They are not duplicated. They both have to check whether the `_selectedIndex` is `null` or not at first and the rest is their condition. IMO I don't think making a flag here could a lot of difference here. 🙏

**Review by ducbm051291 — APPROVED** [review]

File: lib/screens/detail/nps_answer.dart [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                _localizations.surveyNpsNotAtAllLikely,
+                style: _textTheme.labelMedium?.copyWith(color: () {
+                  if (_selectedIndex != null &&
+                      (_selectedIndex ?? 0) <
+                          widget.answers.length / 2.round()) {
+                    return Colors.white;
+                  } else {
+                    return Colors.white.withOpacity(0.5);
+                  }
+                }()),
```
Comment (manh-t) [inline lib/screens/detail/nps_answer.dart:58]: @kaungkhantsoe sorry but we don't allow writing an anonymous function directly to get a value like this. It's not friendly and makes us confused when looking at the code.

Normally we have two approaches:
- Extract to a function such as `getNpsTextColor`.
- Apply condition like this:
```suggestion
                style: _textTheme.labelMedium?.copyWith(
                  color: (_selectedIndex != null &&
                          (_selectedIndex ?? 0) <
                              widget.answers.length / 2.round())
                      ? Colors.white
                      : Colors.white.withOpacity(0.5),
                ),
```

File: lib/screens/detail/nps_answer.dart [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                _localizations.surveyNpsExtremelyLikely,
+                style: _textTheme.labelMedium?.copyWith(color: () {
+                  if (_selectedIndex != null &&
+                      (_selectedIndex ?? 0) >=
+                          (widget.answers.length / 2).round()) {
+                    return Colors.white;
+                  } else {
+                    return Colors.white.withOpacity(0.5);
+                  }
+                }()),
```
Comment (manh-t) [inline lib/screens/detail/nps_answer.dart:71]: same here, please refactor it

File: lib/screens/detail/nps_answer.dart [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                topLeft: radius,
+                bottomLeft: radius,
+              );
+            } else if (isEndItem) {
+              return const BorderRadius.only(
+                bottomRight: radius,
+                topRight: radius,
+              );
+            }
+          }(),
```
Comment (manh-t) [inline lib/screens/detail/nps_answer.dart:111]: and here

File: lib/screens/detail/nps_answer.dart [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            style: () {
+              if (_selectedIndex != null && index <= (_selectedIndex ?? 0)) {
+                return _textTheme.labelLarge;
+              } else {
+                return _textTheme.labelLarge?.copyWith(
+                  fontWeight: FontWeight.w400,
+                  color: Colors.white.withOpacity(0.5),
+                );
+              }
+            }(),
```
Comment (manh-t) [inline lib/screens/detail/nps_answer.dart:125]: here too

**Review by manh-t — CHANGES_REQUESTED** [review]

**Comment by kaungkhantsoe** [comment]
@manh-t  Fixed all in https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/89/commits/0e8cdd84aab9fad10397e36dfb2e2e7b2ad17708

**Review by manh-t — APPROVED** [review]

**Pull request merged**